drivers/sdhc/Kconfig.intel

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+config INTEL_EMMC_HOST
+	bool "EMMC driver"
+	select SDHC_SUPPORTS_NATIVE_MODE
+	select EVENTS
+	default y
+	depends on DT_HAS_INTEL_EMMC_HOST_ENABLED
+	help
+	  EMMC driver support. Though this driver can be easly port to
+	  any HW which is complain to eMMC spec, currently it is only
+	  validated using intel's EMMC host controller.
+
+if INTEL_EMMC_HOST
+config INTEL_EMMC_HOST_INTR
+	bool "EMMC host controller interrupt mode"
+	default y
+	help
+	  EMMC host controller interrupt mode support.
+
+config INTEL_EMMC_HOST_DMA
+	bool "EMMC host controller DMA mode"
+	select DCACHE
+	help
+	  EMMC host controller DMA mode support.
+
+config INTEL_EMMC_HOST_ADMA
+	bool "EMMC host controller ADMA mode"
+	depends on INTEL_EMMC_HOST_DMA
+	help
+	  EMMC host controller ADMA mode support.
+
+config INTEL_EMMC_HOST_ADMA_DESC_SIZE
+	int "EMMC host controller ADMA Descriptor size"
+	default 32
```

Comment:
> Since we use IS_ENABLED this might cause build error during compile time

It seems you're then also waisting memory, since `struct emmc_data` will always contain the `desc_table` even if ADMA is not enabled. It might be worth looking into different ways of fixing this. Some options come to mind:
- Stop using IS_ENABLED
- Refactor the code so the memory is allocated in a separate struct
- Use something like this:
```c
#ifdef CONFIG_INTEL_EMMC_HOST_ADMA_DESC_SIZE
#define ADMA_DESC_SIZE CONFIG_INTEL_EMMC_HOST_ADMA_DESC_SIZE
#else
#define ADMA_DESC_SIZE 0
#endif
```

For the record, the current code works fine, so this could also be done as a follow-up commit (especially considering that time is running low on getting a new driver merged before the feature freeze)

Reply:
updated as per your suggestion